src/main/java/de/adesso/kicker/statistics/trackedranking/persistence/TrackedRanking.java

```diff
@@ -0,0 +1,34 @@
+package de.adesso.kicker.statistics.trackedranking.persistence;
+
+import de.adesso.kicker.user.persistence.User;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import javax.persistence.*;
+import java.time.LocalDate;
+
+@Data
+@NoArgsConstructor
+@Entity
+public class TrackedRanking {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private String rankingId;
+
+    private int rank;
+
+    private int rating;
+
+    @OneToOne(fetch = FetchType.LAZY)
```

Comment:
Isnt ManyToOne needed here?

Reply:
No because one instance of a tracked ranking is tracking one user

Reply:
Well yes but each instance tracks one user

Reply:
so would'nt the method in repo then needed to be called `findByUser()` ?

Reply:
Also the trackedranking doesn't contain lists - it is more like a daily snapshot of the user stats

Reply:
No because that would require use to be a unique key which it isn't and it would only return a single instance.

Reply:
one user can have many ranking objects (one per day)
one ranking object can have one user

Reply:
Woudn't @OneToOne prevent that a user can 'have' multiple Tracker?

Reply:
No because each **instance** tracks only one user

Reply:
And a user has only one ranking object

Reply:
> 
> 
> No because each **instance** tracks only one user

With OneToOne you are limiting the number of instances per user to one

Reply:
I tested it and multiple instances of `TrackedRanking` were created.

Reply:
if there would be only one TrakedRanking object per user it would need to have lists

Reply:
And it isn't a bidirectional relationship. Each TrackedRanking has one user. User's don't know anything about 'TrackedRanking'

Reply:
This is from the hibernate docs on Unidirectional one to one relationships:
"From a relational database point of view, the underlying schema is identical to the unidirectional @ManyToOne association, as the client-side controls the relationship based on the foreign key column."

Reply:
We both discussed this with our co-workers, think about doing a bidirectional relationship as deleting a user would lead to a dead end in the TrackedRanking.

Reply:
When trying to delete a user nothing would happen as this would lead to a violation of the referential integrity. And doing a bidirectional relationship would cause us to have a circular dependency.

Reply:
I'm a bit confused right now, maybe you're right, but I'm definetly too un shure to approve this PR right now. Maybe I get this tommo.

Reply:
Understandable I'm a bit unsure myself. Also you couldn't approve this anyway because this is a *draft* PR which wasn't meant to be reviewed yet.

Reply:
The mapping is now a `@ManyToOne` relationship with `@OnDelete(action = OnDeleteAction.CASCADE)` to delete it when the parent entity, in this case the user, is deleted.

Reply:
Nice, if you're still unsure you can look it up in 02 Konzeptionelles Modell - page 32